### PR TITLE
API for github team updates

### PR DESCRIPTION
**Pull request by isuruf**
This API is provided for use in conda-forge-webservices

### CODE REVIEW

**Comment by jakirkham** [comment]
What remains to be done here just so that I know.

**Comment by isuruf** [comment]
Testing. (Writing a test or if that's hard manually testing)

**Comment by jakirkham** [comment]
Ok, but as far as the API and implementation are concerned, do you feel this is in a good place or was there more that we should be trying to tackle there?

**Comment by isuruf** [comment]
This is okay now. These functions are going to be used in conda-forge-webservices to do welcome messages, etc.

File: conda_smithy/github.py [hunk 1]
```diff
@@ -116,64 +124,73 @@ def create_github_repo(args):
...
+    for new_maintainer in maintainers - current_maintainers_handles:
+        add_membership(team, new_maintainer)
+
+    # Mention any maintainers that need to be removed (unlikely here).
+    for old_maintainer in current_maintainers_handles - maintainers:
+        print(
+            "AN OLD MEMBER ({}) NEEDS TO BE REMOVED FROM {}".format(
+                old_maintainer, repo_name
+            )
+        )
```
Comment (jakirkham) [inline conda_smithy/github.py]: Are we planning to handle removals here. Should add that we should be standardize casing all usernames before this check as GitHub is case insensitive and tends to switch to lower. So some might not match even though they are in both groups.

Comment (isuruf) [reply conda_smithy/github.py]: Both are lower-cased. See https://github.com/conda-forge/conda-smithy/pull/565/files#diff-4970770929969f2ad4e818ed6338e5b1R146 and https://github.com/conda-forge/conda-smithy/pull/565/files#diff-4970770929969f2ad4e818ed6338e5b1R166

**Comment by isuruf** [comment]
Ping

**Pull request merged**